src/main/java/uk/gov/pay/directdebit/payments/model/DirectDebitStatesGraph.java

```diff
@@ -0,0 +1,54 @@
+package uk.gov.pay.directdebit.payments.model;
+
+import com.google.common.graph.ImmutableValueGraph;
+import com.google.common.graph.MutableValueGraph;
+import uk.gov.pay.directdebit.mandate.model.MandateState;
+import uk.gov.pay.directdebit.mandate.model.MandateStatesGraph;
+import uk.gov.pay.directdebit.payments.exception.InvalidStateTransitionException;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static uk.gov.pay.directdebit.mandate.model.MandateState.PENDING;
+
+public abstract class DirectDebitStatesGraph<T extends DirectDebitState> {
```

Comment:
Nice 👍 ~148~ 0 days since use of generics :D

Reply:
Is this a good thing? 😄